crates/dbs-miniball/src/vmm/src/lib.rs

```diff
@@ -0,0 +1,1203 @@
+// Copyright 2022 Alibaba Cloud. All Rights Reserved.
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+//! Miniball built with dragonball-sandbox and rust-vmm components and minimal glue.
+#![deny(missing_docs)]
+
+use std::convert::TryFrom;
+#[cfg(target_arch = "aarch64")]
+use std::convert::TryInto;
+use std::fs::File;
+use std::io::{self, stdin, stdout};
+use std::ops::DerefMut;
+use std::path::PathBuf;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::{Arc, Mutex};
+
+use event_manager::{EventManager, EventOps, Events, MutEventSubscriber, SubscriberOps};
+use kvm_bindings::KVM_API_VERSION;
+use kvm_ioctls::{
+    Cap::{self, Ioeventfd, Irqchip, Irqfd, UserMemory},
+    Kvm,
+};
+use linux_loader::cmdline;
+#[cfg(target_arch = "x86_64")]
+use linux_loader::configurator::{
+    self, linux::LinuxBootConfigurator, BootConfigurator, BootParams,
+};
+#[cfg(target_arch = "x86_64")]
+use linux_loader::{bootparam::boot_params, cmdline::Cmdline};
+
+use linux_loader::loader::{self, KernelLoader, KernelLoaderResult};
+#[cfg(target_arch = "x86_64")]
+use linux_loader::loader::{
+    bzimage::BzImage,
+    elf::{self, Elf},
+    load_cmdline,
+};
+use vm_device::bus::{MmioAddress, MmioRange};
+#[cfg(target_arch = "x86_64")]
+use vm_device::bus::{PioAddress, PioRange};
+use vm_device::device_manager::IoManager;
+#[cfg(target_arch = "aarch64")]
+use vm_device::device_manager::MmioManager;
+#[cfg(target_arch = "x86_64")]
+use vm_device::device_manager::PioManager;
+#[cfg(target_arch = "aarch64")]
+use vm_memory::GuestMemoryRegion;
+use vm_memory::{GuestAddress, GuestMemory, GuestMemoryMmap};
+#[cfg(target_arch = "x86_64")]
+use vm_superio::I8042Device;
+#[cfg(target_arch = "aarch64")]
+use vm_superio::Rtc;
+use vm_superio::Serial;
+use vmm_sys_util::{epoll::EventSet, eventfd::EventFd, terminal::Terminal};
+
+#[cfg(target_arch = "x86_64")]
+use boot::build_bootparams;
+pub use config::*;
+use devices::virtio::block::{self, BlockArgs};
+use devices::virtio::net::{self, NetArgs};
+use devices::virtio::{Env, MmioConfig};
```

Comment:
These module have not been implemented?
Is that possible to mock?
Or you want to have this here and implement this later?